src/equation_of_state.cpp

```diff
@@ -1837,20 +1842,240 @@ MFEM_HOST_DEVICE void PerfectMixture::computeSheathBdrFlux(const double *state,
       double VB = sqrt((T_h + Zsp * T_e) * UNIVERSALGASCONSTANT / msp);
       // TODO(kevin): need to check the sign (out of wall or into the wall)
       bcFlux.primFlux[sp] = VB;
-      bcFlux.primFlux[numSpecies - 2] += Zsp * n_sp[sp] * VB;
-      bcFlux.primFlux[numSpecies - 1] -= msp * n_sp[sp] * VB;  // fully catalytic wall.
+      bcFlux.primFlux[iElectron] += Zsp * n_sp[sp] * VB;
+      bcFlux.primFlux[iBackground] -= msp * n_sp[sp] * VB;  // fully catalytic wall.
     }
   }
-  bcFlux.primFlux[numSpecies - 2] /= n_sp[numSpecies - 2];
-  bcFlux.primFlux[numSpecies - 1] -=
-      GetGasParams(numSpecies - 2, GasParams::SPECIES_MW) * n_sp[numSpecies - 2] * bcFlux.primFlux[numSpecies - 2];
-  bcFlux.primFlux[numSpecies - 1] /= GetGasParams(numSpecies - 1, GasParams::SPECIES_MW) * n_sp[numSpecies - 1];
+  bcFlux.primFlux[iElectron] /= n_sp[iElectron];
+  bcFlux.primFlux[iBackground] -=
+      GetGasParams(iElectron, GasParams::SPECIES_MW) * n_sp[iElectron] * bcFlux.primFlux[iElectron];
+  bcFlux.primFlux[iBackground] /= GetGasParams(iBackground, GasParams::SPECIES_MW) * n_sp[iBackground];
 
   if (twoTemperature_) {
-    double vTe = sqrt(8.0 * UNIVERSALGASCONSTANT * T_e / PI / GetGasParams(numSpecies - 2, GasParams::SPECIES_MW));
-    double gamma = -log(4.0 / vTe * bcFlux.primFlux[numSpecies - 2]);
+    double vTe = sqrt(8.0 * UNIVERSALGASCONSTANT * T_e / PI / GetGasParams(iElectron, GasParams::SPECIES_MW));
+    double gamma = -log(4.0 / vTe * bcFlux.primFlux[iElectron]);
 
     bcFlux.primFlux[numSpecies + nvel_ + 1] =
-        bcFlux.primFlux[numSpecies - 2] * (gamma + 2.0) * n_sp[numSpecies - 2] * UNIVERSALGASCONSTANT * T_e;
+        bcFlux.primFlux[iElectron] * (gamma + 2.0) * n_sp[iElectron] * UNIVERSALGASCONSTANT * T_e;
+  }
+}
+
+
+
+
+void PerfectMixture::GetSpeciesFromLTE(double *conserv, double *primit, TableInterpolator2D *energy_table,
+              TableInterpolator2D *R_table, TableInterpolator2D *c_table, TableInterpolator2D *T_table)  {
+  for (int sp = 0; sp < numActiveSpecies; sp++) {
+    conserv[nvel_ + 2 + sp] = 0.0;
+    primit[nvel_ + 2 + sp] = 0.0;
+  }
+
+  Vector n_sp(numSpecies);    // Define number densities temporarily as [mol/m^3]
+                              // In the end, we convert conserved states for species back to [kg/m^3]
+
+  double rho = 0.0;  // Density of the mixture
+
+  // Is it better to define these in the header file?
+  double T_h = 0.0, T_e = 0.0, p_0 = 0.0;
+  double n_0 = 0.0, n_e = 0.0, n_neutral = 0.0;
+  double lambda_e = 0.0;
+  double tempSaha = 0.0, tempRatio = 0.0;
+  double Q_n = 0.0, Q_i = 0.0;
+  int nPop = 0;
+
+  const double qeOverkB = ELECTRONCHARGE / BOLTZMANNCONSTANT;
+
+
+  // Indexing:
```

Comment:
Is this just an example or is the function specific to the 6 species argon case?

Reply:
Comments have been added regarding the generality of the routine.